content/articles/encrypting-and-decrypting-using-rijndael-key-in-c#/index.md

```diff
@@ -0,0 +1,193 @@
+
+#### Decryption
+Decryption involves the conversion of ciphertext back to original data for authorized access.`CreateDecryptor()` function uses a key to decrypt data. String keys are passed through `CreateEncryptor()` function.
+
+```C#
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Security.Cryptography;
+namespace string_decryption
+{
+    public static class Decryption
+    {
+        public static string Decode()
+        {
+            try
+            {
+                string ourText = "bIwl1Y/220KPke+MzXxKErqBT5jkOEbDJnVEPxCzs7I=";
+                string x = null;
+                string _key = "abcdefgh";
+                string privatekey = "hgfedcba";
+                byte[] privatekeyByte = { };
+                privatekeyByte = Encoding.UTF8.GetBytes(privatekey);
+                byte[] _keybyte = { };
+                _keybyte = Encoding.UTF8.GetBytes(_key);
+                byte[] inputtextbyteArray = new byte[ourText.Replace(" ", "+").Length];
+                //This technique reverses base64 encoding when it is received over the Internet.
+                inputtextbyteArray = Convert.FromBase64String(ourText.Replace(" ", "+"));
+                using (DESCryptoServiceProvider dEsp = new DESCryptoServiceProvider())
+                {
+                    var memstr = new MemoryStream();
+                    var crystr = new CryptoStream(memstr, dEsp.CreateDecryptor(_keybyte, privatekeyByte), CryptoStreamMode.Write);
+                    crystr.Write(inputtextbyteArray, 0, inputtextbyteArray.Length);
+                    crystr.FlushFinalBlock();
+                    return Encoding.UTF8.GetString(memstr.ToArray());
+                }
+                return x;
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
+        static void Main(string[] args)
+        {
+            string donedecrypting = Decode();
+            Console.Write(donedecrypting);
+        }
+    }
+}
+```
+
+Output:
+
+```bash
+section engineering education
+```
+
+The encrypted string `bIwl1Y/220KPke+MzXxKErqBT5jkOEbDJnVEPxCzs7I=` is converted to its original meaning `section engineering education` using `publickey` and `secret key` keys.
+
+A `symmetric key` is used to encrypt data and can also be used to decrypt data using only one string.
+
+Symmetric keys known as `Symmetric Cryptography Algorithms` are utilized in data encryption and decryption. Symmetric Cryptography Algorithms use one key and store large data volumes, thus appropriate for use in DES and Rijndael platforms. `Asymmetric keys` use two keys for encryption and decryption(private and public keys). The public key encrypts data, whereas the private key decrypts data. `RSA` and `digital` signatures` employ asymmetric encryption.

```

Comment:
Try to enhance this section, also use the code snippets correctly (` employ asymmetric encryption)